### PR TITLE
Update `Slice` to use `Bytes` internally

### PR DESCRIPTION
**Pull request by LucioFranco**
This is somewhat incomplete as we do some copying and tests that use `SkipList` create a seg fault but thought I would open a draft.

Btw, super cool project :)

### CODE REVIEW

**Comment by Fullstop000** [comment]
Thank you for your contribution @LucioFranco. This is huge! And maybe we can just use `pub struct Slice(Bytes)`?

**Comment by LucioFranco** [comment]
We could use a tuple struct, honestly, its probably better to just expose bytes externally maybe, though that would be a pretty big change.

I am also unsure why there is a segfault that happens when adding an empty key. Do you have any idea?

**Comment by Fullstop000** [comment]
@LucioFranco I'll investigate this. There are plenty of unsafe codes that can make this happen :(.

File: src/util/slice.rs [hunk 1]
```diff
@@ -30,22 +31,24 @@ use std::slice;
...
-    size: usize,
+    data: Bytes,
 }
 
 impl Slice {
     pub fn new(data: *const u8, size: usize) -> Self {
-        Self { data, size }
+        let data = if !data.is_null() {
+            let slice = unsafe { slice::from_raw_parts(data, size) };
+            Bytes::copy_from_slice(&slice[..])
```
Comment (Fullstop000) [inline src/util/slice.rs]: It seems we can't avoid the copying. And the original version just acquire a ptr.

Comment (LucioFranco) [reply src/util/slice.rs]: Yeah, it is unfortunate, though I am curious if we can just replace all usage of raw slices with bytes? I took a quick look and it seems that there are only a few places that its required.

**Comment by Fullstop000** [comment]
@LucioFranco All the `Slice`s are removed now!

**Pull request closed without merging**